File: backend/app/schemas/goal.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class GoalCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: str | None = Field(None, max_length=1000)
    privacy: Literal["friends", "private"] = "friends"
    streak_interval: int | None = None


class GoalResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    description: str | None
    completed: bool
    archived: bool
    privacy: str
    streak_count: int
    streak_interval: int | None
    last_posted_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontendnew/Streakd/Models/Goal.swift
import Foundation

struct Goal: Codable, Identifiable, Equatable, Hashable {
    let id: String
    let userId: String
    let title: String
    let description: String?
    var completed: Bool
    var archived: Bool
    let privacy: String
    var streakCount: Int
    let streakInterval: Int?
    let notificationTime: String?
    var lastPostedAt: String?
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, title, description, completed, archived, privacy
        case userId = "user_id"
        case streakCount = "streak_count"
        case streakInterval = "streak_interval"
        case notificationTime = "notification_time"
        case lastPostedAt = "last_posted_at"
        case createdAt = "created_at"
    }
}

struct CreateGoalRequest: Encodable {
    let title: String
    let description: String?
    let privacy: String
    let streakInterval: Int

    enum CodingKeys: String, CodingKey {
        case title, description, privacy
        case streakInterval = "streak_interval"
    }
}
